lxd/storage.go

```diff
@@ -187,24 +200,20 @@ func storageForFilename(d *Daemon, filename string) (storage, error) {
 	config := make(map[string]interface{})
 	storageType := storageTypeDir
 
-	filesystem, err := filesystemDetect(filename)
-	if err != nil {
-		return nil, fmt.Errorf("couldn't detect filesystem for '%s': %v", filename, err)
-	}
-
-	lvLinkPath := filename + ".lv"
-	if shared.PathExists(lvLinkPath) {
+	if shared.PathExists(filename + ".lv") {
 		storageType = storageTypeLvm
-		lvPath, err := os.Readlink(lvLinkPath)
+		lvPath, err := os.Readlink(filename + ".lv")
 		if err != nil {
-			return nil, fmt.Errorf("couldn't read link dest for '%s': %v", lvLinkPath, err)
+			return nil, fmt.Errorf("couldn't read link dest for '%s': %v", filename+".lv", err)
 		}
 		vgname := filepath.Base(filepath.Dir(lvPath))
 		config["vgName"] = vgname
-
-	} else if filesystem == "btrfs" {
+	} else if shared.PathExists(filename + ".btrfs") {
```

Comment:
Oh yeah, I'll revert that part then... Might have been nice to do the same .btrfs => symlink trick I did for Zfs so it's more visible to the user, but doing so retroactively is a bit annoying so let's not.